Infrastructure/Modules/CDNHelpers.psm1

```diff
@@ -0,0 +1,204 @@
+function Start-BlobCopy {
+    <#
+.SYNOPSIS
+Copy content to blob storage and update ContentType(MIME) settings
+
+.DESCRIPTION
+Copy content to blob storage and update ContentType(MIME) settings using AzCopy
+AzCopy syntax: https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azcopy
+
+.PARAMETER Source
+The source location of the files to be copied
+
+.PARAMETER Destination
+The blob destinaton of where to copy the files
+
+.PARAMETER SaSToken
+The SaS Token to access the blob storage container
+
+.PARAMETER OriginType
+The Origin Type i.e. "Storage", "Cloud Service", "Web App" or "Custom Origin"
+
+.EXAMPLE
+
+$DeploymentParameters = @ {
+    Source = "c:\FilesToBeCopied\"
+    Destination = "https://name.blob.core.windows.net/cdn"
+    SaSToken = "MySecureSaStokenString"
+    OriginType = "Storage"
+
+}
+BlobCopy @DeploymentParameters
+
+.NOTES
+
+    Suppressed Script Analyzer rules:
+         - PSUseShouldProcessForStateChangingFunctions - The function does not alter the state of an object
+
+#>
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseShouldProcessForStateChangingFunctions", "")]
+    Param(
+        [Parameter(Mandatory = $true)]
+        [String]$Source,
+        [Parameter(Mandatory = $true)]
+        [String]$Destination,
+        [Parameter(Mandatory = $true)]
+        [String]$SaSToken,
+        [Parameter(Mandatory = $true)]
+        [ValidateSet("Storage", "Cloud Service", "Web App", "Custom Origin")]
+        [String]$OriginType
+    )
+    try {
+        if ($OriginType -eq "Storage") {
+            # --- Set location for AzCopy.exe
+            $AzCopyPath = 'C:\Program Files (x86)\Microsoft SDKs\Azure\AzCopy\'
```

Comment:
This needs to be expanded somewhat, we can't guarantee that it will be installed into the same folder on every machine, it may be better to search $ENV:PATH for it instead